src/codeu/chat/client/simplegui/ChatGuiFX.java

```diff
@@ -56,29 +54,61 @@ public void start(Stage primaryStage) throws Exception {
 
         // Sign in page
 
-        this.thestage = primaryStage;            // Initialize the main stage
+        this.thestage = primaryStage;                           // Initialize the main stage
+
+        BorderPane signInPane = new BorderPane();               // Initialize panes
+        FlowPane signInLabelPane = new FlowPane();
+        HBox inputMasterBox = new HBox();
+        inputMasterBox.setSpacing(5);
+        VBox inputVBox = new VBox();
+        HBox usernameHBox = new HBox();
+        HBox passHBox = new HBox();
 
-        signInPane = new BorderPane();                      // Initialize panes
-        signInLabelPane = new FlowPane();
-        signInLabelPane.setAlignment(Pos.BOTTOM_CENTER);       // Make sure label will be in center
+        // Set Pane alignments
 
-        signInLabel = new Label("Sign-in screen");      // Initialize label
-        signInLabel.setFont(Font.font(20));                 // Set style
+        signInLabelPane.setAlignment(Pos.BOTTOM_CENTER);
+        inputMasterBox.setAlignment(Pos.CENTER);
+        inputVBox.setAlignment(Pos.CENTER);
+        usernameHBox.setAlignment(Pos.CENTER);
+        passHBox.setAlignment(Pos.CENTER);
+
+        Label signInLabel = new Label("Sign-in screen");       // Title for sign-in page
+        Label userLabel = new Label("Username:");
+        Label passLabel = new Label("Password:");
+        signInLabel.setFont(Font.font(20));                         // Set style
+        userLabel.setFont(Font.font(15));
+        passLabel.setFont(Font.font(15));
 
         signInButton = new Button("Sign in");                      // Initialize sign in button
-        signInButton.setOnAction((event)-> ButtonClicked(event));       // Initialize its event handler
+        signInButton.setOnAction((event)-> buttonClicked(event));       // Initialize its event handler
+
 
+        userInput = new TextField();
+        passInput = new PasswordField();                        // Set up password fields
```

Comment:
Are there any client-side restrictions enforced by the GUI? Can I register with a username or password of empty string? Can my username contain special characters or sql injection? Can it be a bunch of random unicode that might not display on every machine?

There isn't any one correct answer to this, and you don't have to figure it out right now. But might be something to think about as you hook your GUI up to the controller.